jerry-main/main-unix.c

```diff
@@ -334,6 +349,32 @@ print_unhandled_exception (jerry_value_t error_value) /**< error value */
   jerry_release_value (err_str_val);
 } /* print_unhandled_exception */
 
+/**
+ * Register a JavaScript function in the global object.
+ */
+static void
+register_js_function (const char *name_p, /**< name of the function */
+                      jerry_external_handler_t handler_p) /**< function callback */
+{
+  jerry_value_t global_obj_val = jerry_get_global_object ();
+
+  jerry_value_t function_val = jerry_create_external_function (handler_p);
+  jerry_value_t function_name_val = jerry_create_string ((const jerry_char_t *) name_p);
+  jerry_value_t result_val = jerry_set_property (global_obj_val, function_name_val, function_val);
+
+  jerry_release_value (function_val);
+  jerry_release_value (function_name_val);
+
+  if (jerry_value_has_error_flag (result_val))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_WARNING, "Warning: failed to register '%s' method.", name_p);
+    print_unhandled_exception (result_val);
+  }
+
+  jerry_release_value (result_val);
+  jerry_release_value (global_obj_val);
```

Comment:
Is there a reason why the values are released in this order and in these two phases? If "release everything as soon as possible" is the policy, then reference to the global object can be released after the `jerry_set_property` call. If "release everything as the last step, in reverse order of allocation", then  `result_val`, `function_name_val`, `function_val`, `global_obj_val` should be the order of release at the end of the function.